cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -432,8 +432,9 @@ struct EncPage {
  */
 constexpr bool is_string_col(ColumnChunkDesc const& chunk)
 {
-  return (chunk.data_type & 7) == BYTE_ARRAY and (chunk.data_type >> 3) != 4 and
-         chunk.converted_type != DECIMAL;
+  return ((chunk.data_type & 7) == BYTE_ARRAY and (chunk.data_type >> 3) != 4 and
```

Comment:
if you don't mind, please break this condition into pieces and give them names. It's getting a bit too hard to follow.
Something like:
```
auto const not_converted_to_decimal = chunk.converted_type != DECIMAL;
auto const non_hashed_byte_array = (chunk.data_type & 7) == BYTE_ARRAY and (chunk.data_type >> 3) != 4
auto const fixed_len_byte_array = (chunk.data_type & 7) == FIXED_LEN_BYTE_ARRAY;
return not_converted_to_decimal and (non_hashed_byte_array or fixed_len_byte_array)
```